src/imitation/algorithms/adversarial.py

```diff
@@ -147,65 +160,104 @@ def gen_policy(self) -> BaseRLModel:
     """Policy (i.e. the generator) being trained."""
     return self._gen_policy
 
-  def train_disc(self, n_steps=10, **kwargs):
+  def train_disc(self, total_timesteps: Optional[int] = None) -> None:
     """Trains the discriminator to minimize classification cross-entropy.
 
     Args:
-        n_steps (int): The number of training steps.
-        gen_obs (np.ndarray): See `_build_disc_feed_dict`.
-        gen_acts (np.ndarray): See `_build_disc_feed_dict`.
-        gen_next_obs (np.ndarray): See `_build_disc_feed_dict`.
+      total_timesteps: The number of transitions to sample from the generator
+        replay buffer and the expert demonstration dataset. (Half of the
+        samples are from each source).
+        By default, `self.disc_batch_size`. The number of optimizer updates
+        performed is `total_timesteps // self.n_disc_minibatch`.
+    """
+    if total_timesteps is None:
+      total_timesteps = self.disc_batch_size
+    n_epochs = total_timesteps // self.disc_batch_size
```

Comment:
should check `total_timesteps` is divisible by `self.disc_batch_size`?

Reply:
I think this would be too strict (Stable Baselines just does `total_timesteps // batch_size` without warning for what it's worth).

At the very least I'll add some stuff to the documentation noting when we are rounding down total_timesteps.

Reply:
In my head I'm thinking of `train_gen` and `train_disc` like I think of `BaseRLModel.learn(total_timesteps)`. You can stick any positive int inside there you want.

Might not be the right model

Reply:
`train_gen` and `train_disc` are really just performing a single training step though, whereas Stable Baselines `train()` is meant to train for many epochs. Rounding from 1 million to 999424 isn't a big deal, but 2000 to 1024 is.

Reply:
ok, made this strict.